sonarts-core/tests/rules/noDeadStoreRule/noDeadStoreRule.lint.ts

```diff
@@ -96,3 +94,22 @@ class MyClass {
   }
 
 }
+
+function nextedClassExpression(param: any) {
+  // OK, used in nested class
+  const castedParam = param as any;
+  return class { innerProperty = castedParam; };
```

Comment:
@stas-vilchik It's more interesting to show that even a `write` in an inner class is enough to make castedParam not raise a deadStore. This and the other tests could be read as "since we read castedParam in the nested class we are fine", but it's more general than that.